util/dvsim/LocalLauncher.py

```diff
@@ -66,39 +76,55 @@ def _do_launch(self):
     def poll(self):
         '''Check status of the running process
 
-        This returns 'D', 'P' or 'F'. If 'D', the job is still running. If 'P',
-        the job finished successfully. If 'F', the job finished with an error.
+        This returns 'D', 'P', 'F', or 'K'. If 'D', the job is still running.
+        If 'P', the job finished successfully. If 'F', the job finished with
+        an error. If 'K' it was killed.
 
         This function must only be called after running self.dispatch_cmd() and
         must not be called again once it has returned 'P' or 'F'.
         '''
 
         assert self.process is not None
         if self.process.poll() is None:
+            if self.timeout_secs:
+                if self.start_time + self.timeout_secs < datetime.datetime.now(:
+                    self._kill()
```

Comment:
@matutem, I just realized there could be a problem with having an external entity killing the job on timeout, if we factor in licensing. We specify `-licqueue` switch to both, VCS and Xcelium, which means if a large number tests are run in a limited-license environment, the job will get launched, by the simulator will not start the simulation until a license is acquired. It is possible that `dvsim` may end up killing some tests not because it took long, but because it was stuck waiting for a license. 

I wrote a small snippet of code to instrument the timeout within the simulation itself:
https://www.edaplayground.com/x/7cEr

The `job_timeout` parameter is passed to the simulation via plusarg, and a timeout monitor periodically records the system time every 100ms (arbitrarily picked). On exceeding the timeout, it issues a `$error` message and invokes `$finish` to gracefully kill the simulation. The nice thing about this solution is it is completely independent of `Launcher` variations. We will not need to instrument any change to dvsim since it is a plusarg which will be wired to the already existing `run_opts`. The downsides are - (1): it applies only to simulations. `CompileSim` and `CompileOneShot` will still need a dvsim instrumented `build_timeout` mechanism (2): the in-simulation timeout 'gets' the system time via DPI-C call every 100ms, which itself adds some performance cost. 

LMKWYT - I am fine either way.